packages/textfield/src/mwc-textfield-base.ts

```diff
@@ -0,0 +1,261 @@
+/**
+@license
+Copyright 2019 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import {FormElement, html, query, property, classMap, addHasRemoveClass} from '@material/mwc-base/form-element.js';
+import MDCTextFieldFoundation from '@material/textfield/foundation.js';
+import {MDCTextFieldAdapter} from '@material/textfield/adapter.js';
+import {MDCFloatingLabel} from '@material/floating-label';
```

Comment:
Let's not pull in the MDC component for floating-label, line-ripple, or notched-outline. Instead, we should use use Foundations and Adapters.